datacube_alchemist/cli.py

```diff
@@ -220,22 +220,39 @@ def add_ids_to_queue(config_file, queue, dryrun, ids):
 
 
 @cli.command()
+@click.option(
+    "--predicate",
+    help='Python predicate to filter datasets. Dataset is available as "d"',
+)
 @config_file_option
 @queue_option
 @dryrun_option
-def add_missing_to_queue(config_file, queue, dryrun):
+def add_missing_to_queue(config_file, queue, predicate, dryrun):
     """
     Search for datasets that don't have a target product dataset and add them to the queue
+
+    If a predicate is supplied, datasets which do not match are filtered out.
+
+    Example predicate:
+     - 'd.metadata.gqa_iterative_mean_xy <= 1'
     """
 
     alchemist = Alchemist(config_file=config_file)
 
-    n_messages = alchemist.find_fill_missing(queue, dryrun)
+    datasets = alchemist.find_unprocessed_datasets(queue, dryrun)
+
+    if predicate:
+        code_obj = compile(predicate, "<string>", "eval")
+        datasets = [d for d in datasets if eval(code_obj)]
+        _LOG.info(f'After filtering with "{predicate}", {len(datasets)} remain.')
 
     if not dryrun:
-        _LOG.info(f"Pushed {n_messages} items.")
+        alchemist.datasets_to_queue(queue, datasets)
+        _LOG.info(f"Pushed {len(datasets)} items.")
     else:
-        _LOG.info(f"DRYRUN! Would have pushed {n_messages} items.")
+        _LOG.info(f"DRYRUN! Would have pushed {len(datasets)} alchemist tasks.")
+        for dataset in datasets:
+            _LOG.info(f"Transform: {alchemist.transform_name}; Dataset: {dataset}")
```

Comment:
This is potentially extensive. Logging 20,000 lines is maybe too much. Should we log the first 5?

Reply:
I'd consider making the logging optional, but I like seeing exactly what's going to happen. It can always be redirected to a file for parsing later.

I tried to use `parsed_search_expressions`, but that required rewriting the SQL as SQLAlchemy. I gave up after spending multiple hours on it. Either that, or it would be converting the parsed expressions to Python, and that just seemed way more complicated.